components/builder-db/src/migrations/2020-02-12-195406_audit_origin_table/up.sql

```diff
@@ -0,0 +1,12 @@
+CREATE SEQUENCE IF NOT EXISTS audit_origin_id_seq;
+CREATE TYPE origin_operation AS ENUM ('origin_create', 'origin_delete', 'owner_transfer');
+
+CREATE TABLE IF NOT EXISTS audit_origin (
+    id bigint DEFAULT next_id_v1('audit_origin_id_seq') PRIMARY KEY NOT NULL,
+    operation origin_operation,
+    target_object text,
```

Comment:
:thinking:  I had to pause when I looked at this in the db the first time.  It can store either an origin or an account id, which may be ok, I'll defer to @chefsalim and @eeyun  on that,  but I wonder if this should be nullable and only store target_account_id?  Does it provide value to store the origin here?

Reply:
Great question! I originally had a more extensive list of operation types such as: `IntegrationDelete`, `SecretDelete`, `ChannelDelete`.  I have since removed those additional operations, but I could certainly see them (or others) being required in the future. For that reason, I chose a `target_object` field to hold the value of the "thing" being operated open. Does that answer your question?

Reply:
I think so.  I have vague memories of being bitten by Uber-columns in a previous life, but the alternatives weren't any better.  But that's something we can address when we get there :smiley_cat: